types/layer.go

```diff
@@ -93,7 +94,9 @@ func buildEnv(passThrough []string, newEnv map[string]string,
 }
 
 func (l *Layer) BuildEnvironment() (map[string]string, error) {
-	return buildEnv(l.BuildEnvPt, l.BuildEnv, os.Environ)
+	env, err := buildEnv(l.BuildEnvPt, l.BuildEnv, os.Environ)
+	env["STACKER_LAYER_NAME"] = l.Name
```

Comment:
Is there a reason to have this separately from STACKER_STACKER_DIR and friends? I don't see any.

Reply:
STACKER_STACKER_DIR are "stacker config"  level items.  this is a "layer" level item.  So substitution doesn't really make sense.

```yaml
my-build1:
  run: |
    this is ${STACKER_LAYER_NAME} of my-build1
my-build2:
  run: |
     here it would be my-build2
```

Reply:
Ah, yep, makes sense.